src/Raven.Server/RavenServer.cs

```diff
@@ -1197,6 +1196,25 @@ protected async Task<byte[]> RefreshViaLetsEncrypt(CertificateUtils.CertificateH
                 return null;
             }
 
+            // TODO: force renew should generate new date everytime (abusing this can hit the let's encrypt limits)? or try to reuse the generated certificate for today?
+            var name = $"Refreshed@{renewalDate:O}";
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            using (ctx.OpenReadTransaction())
+            {
+                // let's check if we already did a refresh today
+                // this is a HACK we use the name as a thumbprint here
```

Comment:
I don't like this very much.
Why are we _persisting_ this state?
If we keep this in memory, isn't that enough?

Reply:
yeah, probably in memory is good enough.